src/Elastic.Apm/Report/PayloadSenderV2.cs

```diff
@@ -55,6 +56,9 @@ public PayloadSenderV2(IApmLogger logger, IConfigurationReader configurationRead
 			servicePoint.ConnectionLimit = 20;
 
 			_httpClient = new HttpClient(handler ?? new HttpClientHandler()) { BaseAddress = serverUrlBase };
+			_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue($"elasticapm-{Consts.AgentName}", _service.Agent.Version));
+			_httpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue("system.net.http",
+				typeof(HttpClient).Assembly.GetCustomAttribute<AssemblyFileVersionAttribute>().Version));
```

Comment:
I think you forgot to append `language`/`version` (`Elastic.Apm.Api.Runtime.Name`/`Elastic.Apm.Api.Runtime.Version`) with the adaptation as you proposed at https://github.com/elastic/apm-agent-dotnet/pull/397#discussion_r305664084

Reply:
I left it to last and decided to replace all invalid characters by `"[ /()<>@,:;={}?\\[\\]\"\\\\]"` regex because it doesn't look complicated but guaranteed valid header